crates/auth/src/providers/github/mod.rs

```diff
@@ -0,0 +1,69 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: Apache-2.0
+pub mod routes;
+
+use crate::{error::Error, session::Session};
+
+use http::header::USER_AGENT;
+use oauth2::{basic::BasicClient, AuthUrl, ClientId, ClientSecret, RedirectUrl, TokenUrl};
+use serde::Deserialize;
+
+pub const LOGIN_URI: &str = "/auth/github";
+pub const AUTHORIZED_URI: &str = "/auth/github/authorized";
```

Comment:
The end result is probably that we have `<url>/_auth/{github,google,...}/*`

I don't think `auth` crate should hard-code the routes from the top-level. It *should*, however, define the paths starting at `_auth`. So paths in this crate should look like `/github/<foo>` or `/google/<bar>`